EIPS/eip-6888.md

```diff
@@ -0,0 +1,88 @@
+---
+eip: 6888
+title: Math checking in EVM
+description: Check for math underflows overflows and division by zero at EVM level
+author: Renan Rodrigues de Souza (@RenanSouza2)
+discussions-to: https://ethereum-magicians.org/t/eip-math-checking/13846
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-04-16
+---
+
+## Abstract
+
+This EIP adds many math checking to EVM arithmetic. The list includes underflows, overflows, division by zero. A new opcode is added to get the flags and clean them.
+
+## Motivation
+
+The importance of math checking in smart contract projects is very clear. It was an OpenZeppelin library and then incorporated in Solidity's default behavior. Bringing this to EVM level can combine both gas efficiency and safety.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Starting from `BLOCK >= HARDFORK_BLOCK`
+
+Two new flags are added to the EVM state: unsigned error (`uerr`) and signed overferror (`serr`).
```

Comment:
When adding new variables to the EVM, it can be useful to specify an existing variable that has the same scope. For example, [EIP-3074](https://eips.ethereum.org/EIPS/eip-3074)'s `authorized` has the same scope as the program counter.